src/controller/java/src/chip/devicecontroller/WriteAttributesCallback.java

```diff
@@ -18,12 +18,33 @@
 package chip.devicecontroller;
 
 import chip.devicecontroller.model.ChipAttributePath;
+import javax.annotation.Nullable;
 
 /** An interface for receiving write response. */
 public interface WriteAttributesCallback {
-  void onError(ChipAttributePath attributePath, Exception e);
 
-  void OnResponse(ChipAttributePath attributePath);
+  /**
+   * OnError will be called when an error occurs after failing to write
+   *
+   * - CHIP_ERROR_TIMEOUT: A response was not received within the expected response timeout.
+   * - CHIP_ERROR_*TLV*: A malformed, non-compliant response was received from the server.
+   * - CHIP_ERROR encapsulating a StatusIB: If we got a non-path-specific
+   *   status response from the server.  In that case,
+   *   StatusIB::InitFromChipError can be used to extract the status.
+   * - CHIP_ERROR*: All other cases.
+   *
+   * @param[in] attributePath   The attribute path field in write response.
```

Comment:
nit: the `[in]` or `[out]` modifiers are not a conventional thing for Javadoc.

Generally the format is something like:

```
/**
  * Summary documentation here.
  * 
  * @param attributePath the path where the error occurred
  * @param e the exception that occurred
  */
```

https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#tag

Reply:
updated, thanks